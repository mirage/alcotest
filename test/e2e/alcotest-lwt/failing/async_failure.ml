let free () =
  print_endline "freeing all resources";
  Lwt.return ()

let test_lwt switch () =
  Lwt_switch.add_hook (Some switch) free;
  Lwt.async (fun () -> failwith "All is broken");
  Lwt.pause ()

let () =
  let open Alcotest_lwt in
  Alcotest_test_helper.wakeup_until_resolved
  @@ run __FILE__
       [
         ( "all",
           [
             test_case "one" `Quick test_lwt;
             test_case "two" `Quick (fun _ () -> Lwt.return_unit);
           ] );
       ]
