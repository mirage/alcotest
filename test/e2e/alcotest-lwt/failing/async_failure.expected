Testing `test/e2e/alcotest-lwt/failing/async_failure.ml'.
This run has ID `<uuid>'.

> [FAIL]    all › one
  [PASS]    all › two

┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]    all › one                                                          │
└──────────────────────────────────────────────────────────────────────────────┘
freeing all resources
[failure] All is broken
          <stacktrace>
          
Logs saved to `<build-context>/_build/_tests/<test-dir>/all/one.output'.
 ──────────────────────────────────────────────────────────────────────────────

Full test results in `<build-context>/_build/_tests/<test-dir>'.
1 failure! in <test-duration>s. 2 tests run.
